components/integrations/bots/bots.jsx

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import * as Utils from 'utils/utils.jsx';
+import BackstageList from 'components/backstage/components/backstage_list.jsx';
+import Constants from 'utils/constants.jsx';
+
+import Bot from './bot.jsx';
+
+export default class Bots extends React.PureComponent {
+    static propTypes = {
+
+        /**
+        *  Map from botUserId to bot.
+        */
+        bots: PropTypes.object.isRequired,
+
+        /**
+        *  Map from botUserId to accessTokens.
+        */
+        accessTokens: PropTypes.object.isRequired,
+
+        /**
+        *  Map from botUserId to owner.
+        */
+        owners: PropTypes.object.isRequired,
+
+        actions: PropTypes.shape({
+
+            /**
+            * Ensure we have bot accounts
+            */
+            loadBots: PropTypes.func.isRequired,
+
+            /**
+            * Load access tokens for bot accounts
+            */
+            getUserAccessTokensForUser: PropTypes.func.isRequired,
+
+            /**
+            * Access token managment
+            */
+            createUserAccessToken: PropTypes.func.isRequired,
+            revokeUserAccessToken: PropTypes.func.isRequired,
+            enableUserAccessToken: PropTypes.func.isRequired,
+            disableUserAccessToken: PropTypes.func.isRequired,
+
+            /**
+            * Load owner of bot account
+            */
+            getUser: PropTypes.func.isRequired,
+
+            /**
+            * Disable a bot
+            */
+            disableBot: PropTypes.func.isRequired,
+
+            /**
+            * Enable a bot
+            */
+            enableBot: PropTypes.func.isRequired,
+        }),
+
+        /**
+        *  Only used for routing since backstage is team based.
+        */
+        team: PropTypes.object.isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            loading: true,
+        };
+    }
+
+    componentDidMount() {
+        this.props.actions.loadBots(
+            Constants.Integrations.START_PAGE_NUM,
+            Constants.Integrations.PAGE_SIZE
+        ).then(
+            (result) => {
+                if (result) {
```

Comment:
Shouldn't this be the following?
```suggestion
                if (result.data) {
```